### PR TITLE
254 copy to clipboard

### LINKED ISSUE #254: Copy reply to Clipboard
As an editor, I would like to copy a reply to the clipboard with one click in order to process the content in other applications

### PR DESCRIPTION
**Pull request by tillprochaska**
Merge #318 first. Close #254 

***

### CODE REVIEW

File: app/components/chat_message/chat_message.js [hunk 1]
```diff
@@ -35,4 +35,24 @@ export default class extends Controller {
...
+
+    navigator.clipboard.writeText(clipboardText).then(() => this.onCopy());
+  }
+
+  onCopy() {
+    this.element.classList.add('ChatMessage--copied');
+
+    setTimeout(() => {
+      this.element.classList.remove('ChatMessage--copied');
+    }, 2000);
```
Comment (tillprochaska) [inline app/components/chat_message/chat_message.js:56]: Move this into a constant…

```suggestion
    setTimeout(() => {
      this.element.classList.remove('ChatMessage--copied');
    }, SUCCESS_MESSAGE_DURATION);
```

**Review by drjakob — APPROVED** [review]

**Comment by FrauCsu** [comment]
Approved

File: app/components/chat_message/chat_message.js [hunk 2]
```diff
@@ -35,4 +37,24 @@ export default class extends Controller {
...
+  copy() {
+    const text = this.textTarget.innerText;
+    const sender = this.data.get('sender-name');
+    let clipboardText = `${sender}: ${text}`;
+
+    if (!sender) {
+      clipboardText = text;
+    }
+
+    navigator.clipboard.writeText(clipboardText).then(() => this.onCopy());
```
Comment (roschaefer) [inline app/components/chat_message/chat_message.js:50]: I learned something new today!

File: app/components/chat_message/chat_message.js [hunk 3]
```diff
@@ -35,4 +37,24 @@ export default class extends Controller {
...
+  }
+
+  onCopy() {
+    this.element.classList.add('ChatMessage--copied');
+
+    setTimeout(() => {
```
Comment (roschaefer) [inline app/components/chat_message/chat_message.js:56]: Can't we implement that with CSS?

**Review by roschaefer — APPROVED** [review]
clipboard

@tillprochaska feel free to refactor the effect with CSS if possible.

File: app/components/chat_message/chat_message.js (hunk 3 above)
Comment (tillprochaska) [reply app/components/chat_message/chat_message.js:56]: Yes, that should be possible, e.g. using different transition delays – however I think that’d make it more opaque/more difficult to understand…

**Pull request merged**